### DISCUSSION TITLE
golang-version updates don't adhere to go constraints declaration

### DISCUSSION THREAD

**Discussion by dnwe** [renovatebot/renovate#16922] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.141.0

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

GHE 3.1.22

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

PR https://github.com/renovatebot/renovate/pull/16541 enabled golang-version updates in the `go.mod` file by default. However, it appears that they don't respect the [constraints](https://docs.renovatebot.com/configuration-options/#constraints) configuration regarding `go` version.

My assumption was that an existing `constraints: { go: "1.17" }` configuration option would have also pinned `golang-version` to bumping no higher than `go1.17` in the `go.mod` files. However, it seems this doesn't currently happen.

A workaround is to manually specify a packageRule, but this requires updating the renovate configuration in multiple places when moving up major go versions in the future:
```
        matchDatasources: ["docker", "golang-version"],
        matchPackageNames: ["go", "golang"],
        allowedVersions: "<1.18.0",
```

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
This work as designed, so it's not a bug.

**Comment by viceice** [comment]
Use dashboard approval to manually allow golang updates

**Comment by echarrod** [comment]
> This work as designed, so it's not a bug.

Is this for the same reason as in [this issue](https://github.com/renovatebot/renovate/discussions/20491)?

> The Go maintainers recommended that we always run using the latest release, due to the fact they maintain backwards compatibility

**Reply by rarkins** [reply]
No, these are separate.

**Reply by echarrod** [reply]
Would it be possible to explain a little more about what:
> This work as designed, so it's not a bug.

means then? 🙏

**Reply by rarkins** [reply]
It means that `constraints.go` was never intended to restrict updates of Go itself. Setting `allowedVersions` in a package rule is the solution, not a workaround.
